azure/services/async/async.go

```diff
@@ -82,9 +110,21 @@ func CreateResource(ctx context.Context, scope FutureScope, client Creator, spec
 		return processOngoingOperation(ctx, scope, client, resourceName, serviceName)
 	}
 
-	// No long running operation is active, so create the resource.
+	// No long running operation is active, so get the existing resource.
+	existingResource, parameters, err := getExistingResourceWithParameters(ctx, scope, client, spec, serviceName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create resource %s/%s (service: %s)", rgName, resourceName, serviceName)
+	} else if parameters == nil && existingResource != nil {
+		// Nothing to do, return existing and don't call CreateOrUpdateAsync
+		scope.V(2).Info("resource up to date", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
+		return existingResource, nil
+	} else {
+		errors.Errorf("failed to create resource %s/%s (service: %s): no parameters and no existing resource found", rgName, resourceName, serviceName)
```

Comment:
this isn't actually an error?

Reply:
I don't think we'd ever reach that block if everything is working correctly. If there are no parameters, then we'd have an existing resource that's up to date, and if we have parameters, then we'd be able to create or update using that. If both `parameters` and `existingResource` are `nil`, then it seems like we'd have violated the specs of the function in our code. Does that sound right to you, or is there a better way to handle that?

Reply:
Right, there are really just 3 cases:

if err != nil {
     // something went wrong, return an error
} else if parameters == nil {
    // nothing to do, return existing resource 
}

// otherwise, keep going and create the resource